frontend/src/metabase/components/CreateDashboardModal.unit.spec.js

```diff
@@ -0,0 +1,172 @@
+    nextDashboardState.name = name;
+  }
+  if (description) {
+    const input = screen.getByLabelText("Description");
+    userEvent.clear(input);
+    userEvent.type(input, description);
+    nextDashboardState.description = description;
+  }
+  return nextDashboardState;
+}
+
+describe("CreateDashboardModal", () => {
+  beforeEach(() => {
+    xhrMock.setup();
+    xhrMock.get("/api/collection", {
+      body: JSON.stringify([
+        {
+          id: "root",
+          name: "Our analytics",
+          can_write: true,
+        },
+      ]),
+    });
+  });
+
+  afterEach(() => {
+    xhrMock.teardown();
+  });
+
+  it("displays empty form fields", () => {
+    setup();
+
+    expect(screen.queryByLabelText("Name")).toBeInTheDocument();
+    expect(screen.queryByLabelText("Name")).toHaveValue("");
+
+    expect(screen.queryByLabelText("Description")).toBeInTheDocument();
+    expect(screen.queryByLabelText("Description")).toHaveValue("");
+
+    expect(screen.queryByText("Our analytics")).toBeInTheDocument();
+
+    expect(
+      screen.queryByRole("button", { name: "Create" }),
+    ).toBeInTheDocument();
+    expect(
+      screen.queryByRole("button", { name: "Create" }),
+    ).toBeInTheDocument();
+  });
+
+  it("can't submit if name is empty", () => {
+    setup();
+    expect(screen.queryByRole("button", { name: "Create" })).toBeDisabled();
+  });
+
+  it("calls onClose when Cancel button is clicked", () => {
+    const { onClose } = setup();
+    fireEvent.click(screen.queryByRole("button", { name: "Cancel" }));
+    expect(onClose).toHaveBeenCalledTimes(1);
+  });
+
+  it("submits a create request correctly", done => {
+    const FORM = {
+      name: "New fancy dashboard",
+      description: "Just testing the form",
+    };
+    setupCreateRequestAssertion(done, FORM);
+    setup({ mockCreateDashboardResponse: false });
+
+    fillForm(FORM);
+    fireEvent.click(screen.queryByRole("button", { name: "Create" }));
+  });
+
+  describe("Cache TTL field", () => {
+    beforeEach(() => {
+      mockCachingEnabled();
+    });
+
+    describe("OSS", () => {
+      it("is not shown", () => {
+        setup();
+        expect(screen.queryByText("More options")).not.toBeInTheDocument();
+        expect(
+          screen.queryByText("Cache all question results for"),
+        ).not.toBeInTheDocument();
+      });
+    });
+
+    describe("EE", () => {
+      beforeEach(() => {
+        PLUGIN_CACHING.cacheTTLFormField = {
+          name: "cache_ttl",
+          type: "integer",
+        };
+      });
+
+      afterEach(() => {
+        PLUGIN_CACHING.cacheTTLFormField = null;
+      });
+
+      it("is not shown", () => {
+        setup();
+        expect(screen.queryByText("More options")).not.toBeInTheDocument();
+        expect(
+          screen.queryByText("Cache all question results for"),
+        ).not.toBeInTheDocument();
+      });
+    });
+  });
+
+  test.todo("navigates back to dashboard after an update");
```

Comment:
mean to leave this?

Reply:
Oops, forgot to remove them, thank you. Couldn't figure out a good way to mock router functions yet